### PR TITLE
Abbs 0.06

### PR DESCRIPTION
**Pull request by gcharest**
- Address EA teams Feedback from version 0.05

### CODE REVIEW

**Comment by gcharest** [comment]
This is a significant set of changes so the more people I can have a look at this, the better.

This version is addressing the comments received after ARC and from the entire EA teams. [Comments](https://014gc.sharepoint.com/:x:/s/strategyBuildvsBuy/EeRAackCfjNHtTJ0RBAVBj4BexP7rFoHJqqfsXG69dTWxg?e=IXDJSn) can be found on the SharePoint site.

**Review by smellems — APPROVED** [review]

**Comment by gcharest** [comment]
@smellems any idea why the PMI link is still failing? I checked the network monitor when loading it and it seemed all good, even delay to process the request... :thinking:

File: _pages/en/strategy-build-vs-buy.md [hunk 1]
```diff
@@ -44,18 +48,18 @@ permalink: /strategy-build-vs-buy.html
...
 
-The intent behind this strategy is to communicate a decision by the Chief Information Officer (CIO) (not yet approved) on a path forward (the Guiding Policy), and what investments are needed to operationalize that decision (the coherent set of actions).
+The intent behind this strategy is to communicate a decision by the ESDC Chief Information Officer (CIO) (not yet approved) on a path forward (the Guiding Policy), and what investments are needed to operationalize that decision (the coherent set of actions).
 
 ### Targeted Stakeholders
 
-This **strategy document** is targeted to stakeholders involved in obtaining an IT Solution as an outcome of an initiative as defined in the Policy on Project and Programme Management (PPPM).
+This **strategy document** is targeted to stakeholders involved in acquiring an IT Solution as an outcome of an initiative as defined in the Policy on Project and Programme Management (PPPM).
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:62]: suggest rewording to "... an IT Solution to help fulfill outcomes of an IT-enabled initiative as defined in a Project or Programme Business Case"

The rationale is that an IT Solution should not be seen as a business outcomes. We are thriving to have lines of businesses better define what they are striving to achieve with IT. This is expected to be defined in a business case (via outcomes and metrics) as per the Directive on Programme Management and the Directive on Project Management (both instruments under the PPPM).

File: _pages/en/strategy-build-vs-buy.md [hunk 2]
```diff
@@ -132,31 +149,48 @@ Stakeholders are expected to still comply with existing policy instruments inclu
...
+Looking for a default answer and approach to the question "Should an IT Solution be built or bought?" at the organizational level is a mental shortcut that bypasses GoC Mandatory Procedures for [Enterprise Architecture Assessments](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32602) and on [Application Programming Interfaces](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32604) as well as the concepts of solution architecture, where the relevant experts of the organization, both from business lines, finance, security, information technology, etc., work together to design a solution while leveraging various resources, including people, tools, processes, and rules.
 
 An IT solution is by default built, whether by a single person, multiple teams or by a third party, and it will most often consist in an assemblage of many components, known as IT Products in this document.
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:196]: Will need to update the diagram to reflect Guiding Policy's Governance, Compliance, and Reporting #1. ("IT Solutions are broken down into manageable parts (IT Products) ")

So change "IT Component" to "IT Product".

File: _pages/en/strategy-build-vs-buy.md [hunk 3]
```diff
@@ -67,48 +71,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
...
+  - ESDC [Target Solution Delivery Model](strategy-target-solution-delivery-model.html) Strategy (draft in consultation)
+  - ESDC Data Strategy
 
 ### Governance, Compliance, and Reporting
 
-1. IT Solutions are broken down into manageable parts (IT Products) in line with the [GoC Business Capability Model](https://gcconnex.gc.ca/file/group/21723432/al#3l3319941)
-2. IT Solutions are assessed against the IT Solutions Risk Assessment Framework (to be created - see [Coherent set of actions](#coherent-set-of-actions) and read [Managing the Risks of Obtaining of Software](#managing-the-risks-of-obtaining-of-software)).
-3. IT Product / IT Solution maintenance and support may be done in-house or outsourced to a 3rd party vendor
-   - Outsourced maintenance and support may be given to a 3rd party different than the one providing the software licence.
+1. IT Solutions are broken down into manageable parts (IT Products) in line with the the ESDC Business Capability Model, whichever is most relevant.
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:123]: The last part of the sentence misses something. "in line with the ESDC business Capability Model, whichever is most relevant".

I think we mean

"... in line with either the ESDC Business Capability Model or the  GoC Business Capability Model, whichever is most relevant"

Also, here is the link to the ESDC Business Capability Model: https://dialogue/grp/EA-AE/SitePages/DBCM.aspx

File: _pages/en/strategy-build-vs-buy.md [hunk 4]
```diff
@@ -67,48 +71,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
...
+2. IT Solutions are assessed against the IT Solutions Risk Assessment Framework (to be created - see [Coherent set of actions](#coherent-set-of-actions) and read [Managing the Risks of Acquiring of Software](#managing-the-risks-of-acquiring-of-software)).
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:124]: link to "Managing the risks of acquiring of software" is broken (tried it in a Jekyll rendered site).

File: _pages/en/strategy-build-vs-buy.md [hunk 5]
```diff
@@ -118,12 +134,13 @@ Stakeholders are expected to still comply with existing policy instruments inclu
...
 ### Architecture
 
-1. IT Solutions and IT products loose coupling boundaries are at the Business Capability Model level.
-   - Duplication of IT Products or IT Solutions is supported as long as they are for different business capabilities.
+1. IT Solutions and IT products loose coupling boundaries are at intersection of value stream, value stage and business capability (ESDC Business Capability Model).
+   - Duplication of IT Solutions or IT Products is supported as long as they are for different business capabilities.
+   - Boundary of IT Solutions and IT Products are to be application capabilities so that they can enable business capabilities.
 2. IT Products are risk assessed for lock-in situations with the IT Products risk assessment framework (to be created - see [Coherent set of actions](#coherent-set-of-actions))
 
 ## Coherent set of actions
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:146]: Question
At Governance / Produce the EARB standard definition of IT Solutions. Do we mean to formally define the relationships between IT Solutions, IT Products, Software, Infrastructure?

i.e. What is in Appendix A - Business Case / Build vs Buy : Oversimplified Question

Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:146]: On item Governance / Update Strategy based on upcoming revision of the Mandatory Procedures on EA and API

Was that a feedback from architecture teams?
It seems redundant to add it as an action for the strategy. If mandatory procedures, it is expected to trigger a review of the strategy.

Suggest removing this action from the list

Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:146]: On Governance / Endorse the ESDC OSS Framework, it says "Officially REVIEW the ESDC OSS Framework and endorse it."

is further review necessary? Should the strategy's action only state "seek endorsement by EARB"?

File: _pages/en/strategy-build-vs-buy.md [hunk 6]
```diff
@@ -132,31 +149,48 @@ Stakeholders are expected to still comply with existing policy instruments inclu
...
+  - Decreased use of IT Solutions and IT Products for which there are non-interoperable and exclusive:
+    - File formats
+    - Application Programming Interfaces
+    - Data exports
+- Building
+  - Increased ratio of internally built projects released as OSS
 - Applicable to all scenarios
-  - Increased speed and flexibility for changes
+  - Increased ratio of decoupled IT Solutions and IT Products compared to baseline
+  - Increased number of APIs accessible for reuse throughout ESDC IT Solutions and IT Products
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:183]: I would rephrase

Increased number of APIs discoverable, and accessible, for reuse throughout ESDC IT solutions and IT Products

File: _pages/en/strategy-build-vs-buy.md [hunk 7]
```diff
@@ -249,21 +347,18 @@ Even traditional services delivered through SaaS, such as email, require conside
 This strategy is intended to return control of implementation of services back to the department thereby permitting increased alignment with the Government of Canada direction, and increase the quality, and rate by which, digital services are delivered to Canadians.
 The intent is to replace dated approaches to IT solutions delivery in favour of practices aligned with industry best practice in order to promote the department's digital transformation and keep pace with the rising expectations of Canadian citizens with respect to digital services.
 
-## Appendix B - Traceability Matrix (Update coming soon)
+## Appendix B - Traceability Matrix
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:350]: Should look to add PwC recommendations in there

File: _pages/en/strategy-build-vs-buy.md [hunk 8]
```diff
@@ -132,31 +149,48 @@ Stakeholders are expected to still comply with existing policy instruments inclu
...
+  - Decreased ratio of IT Solutions and IT Products for which the department can not choose a service integrator and/or a software maintainer other than the group that provides the software licence.
+  - Decreased use of IT Solutions and IT Products for which there are non-interoperable and exclusive:
+    - File formats
+    - Application Programming Interfaces
+    - Data exports
+- Building
+  - Increased ratio of internally built projects released as OSS
 - Applicable to all scenarios
-  - Increased speed and flexibility for changes
+  - Increased ratio of decoupled IT Solutions and IT Products compared to baseline
```
Comment (remyb2canada) [inline _pages/en/strategy-build-vs-buy.md:182]: the challenge here will be to define what is decoupled. Maybe adding it in the definition section (I'm looking at how to define it too so the TSDM should point to this strategy). 

Maybe:

A solution or product is decoupled when:
- it can run independently in production from other product/solutions (at least 24h)
- it's only integration with other product/solutions is via an API contract

**Review by remyb2canada — APPROVED** [review]
Good stuff! 
No show stoppers I see. 
Only providing suggestions

File: _pages/en/strategy-build-vs-buy.md (hunk 1 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:62]: Good point, will adjust!

File: _pages/en/strategy-build-vs-buy.md (hunk 3 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:123]: Yes! Somehow disappeared upon writing that part :thinking: 
Key feedback driving this nuance was from EA teams explaining that some capabilities are specific to ESDC and not identified in the GoC model.

File: _pages/en/strategy-build-vs-buy.md (hunk 4 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:124]: Will fix

File: _pages/en/strategy-build-vs-buy.md (hunk 5 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:146]: I think we might need to do some level of explanations? The TOGAF model is great when you look at the acquisition of IT Solutions (bunch of key concepts to look into) but covers barely anything for the "operations" of that solution. That's mostly covered under ITIL instead where we define the various processes required to ensure SLA are met.

Another dimension we're trying to bring in this version is the breakdown of the rights and conditions we acquire an IT Solution (whether adopted, bought or built) and the actual maintenance (software patches, security fixes, feature enhancements, etc.) of it.

Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:146]: Could adjust to that yes

File: _pages/en/strategy-build-vs-buy.md (hunk 8 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:182]: Will need to add nuance to the concept of IT Solutions Definition itself :thinking:

**Review by gabrielcossette — APPROVED** [review]

File: _pages/en/strategy-build-vs-buy.md (hunk 2 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:199]: Need to update the image, I think you had the original @remyb2canada ?

File: _pages/en/strategy-build-vs-buy.md (hunk 5 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:142]: Probably need to create issue for this.

Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:142]: created issue #1332

Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:142]: Hmm, well it's an action, not a guiding principle. Action to continuously update the strategy :/ Not sure how we can capture that principle of continuously staying aligned.

Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:142]: Removed, kept as mandate in presentation to CIOEC for the team to keep up to date :thinking:

File: _pages/en/strategy-build-vs-buy.md (hunk 8 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md]: I don't have a good answer for the metrics, hard to cautiously identify them. Was asked to add more metrics, and to have less :man_shrugging:

File: _pages/en/strategy-build-vs-buy.md [hunk 9]
```diff
@@ -67,48 +67,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
 
 Moving to the digital age requires improving IT's responsiveness and promoting a shared accountability with ESDC stakeholders on the use of technology.
 
-Yet, the current principle when obtaining IT Solutions is to "Buy what we can, Build what we must."
+Yet, the current principle when acquiring IT Solutions is to "Buy what we can, Build what we must."
 The assumed reasoning behind this overarching principle is that by purchasing an IT Solution from a third party, risks and responsibilities of its operations are handled by a company who has more resources and technical expertise to produce and maintain software.
 
 This principle unfortunately is leading to an oversimplified approach that doesn't take into account that, as a government department, ESDC is subject to multiple and fast-paced changes, whether at the [legal, political, organizational or technological levels](https://www.canada.ca/en/employment-social-development/corporate/reports/departmental-plan/2020-2021.html#h2.2-3.2), that impacts its ability to deliver on its mandate.
```
Comment (RachelMuston) [inline _pages/en/strategy-build-vs-buy.md:73]: I'd remove 'unfortunately' as I don't think its needed. Or if keeping, perhaps put as "this principle is unfortunately leading....'. If staying as is, think commas might be required around unfortunately.

File: _pages/en/strategy-build-vs-buy.md [hunk 10]
```diff
@@ -322,23 +425,27 @@ The following traceability matrix is used to show alignment with various strateg
...
+
 ## Appendix D - Definitions
 
 IT Solution
-: An IT Solution is the combination of 1 or many IT Products, which are in turn comprised of 1 or many Software and/or Hardware, obtained through many possible ways: built internally, obtained as open source, provided by a company as an executable application under a proprietary licence, as a standalone device, or used as a service through a subscription model. (See the following standard IITB definition.)
+: An IT Solution is the combination of 1 or many IT Products, which are in turn comprised of 1 or many Software and/or Hardware, acquired through many possible ways: built internally, acquired as open source, provided by a company as an executable application under a proprietary licence, as a standalone device, or used as a service through a subscription model. (See the following standard IITB definition.)
```
Comment (RachelMuston) [inline _pages/en/strategy-build-vs-buy.md:435]: It seems like we have removed numbers in other areas (3rd to third). So thinking this should read '....the combination of one or many IT Products'.  Also '...which are in turn comprised of one or many Software...'

**Review by RachelMuston — COMMENTED** [review]
Looks awesome Guillaume. My only contribution are some minor edits. :) The strategy is really clear and easy to follow. Wondering... as the strategy has improved (and it really has), is the 'build-vs-buy explained document still needed?  Curious to know what our stakeholders think about that in future engagements.

File: _pages/en/strategy-build-vs-buy.md [hunk 11]
```diff
@@ -67,48 +67,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
...
 
 This principle unfortunately is leading to an oversimplified approach that doesn't take into account that, as a government department, ESDC is subject to multiple and fast-paced changes, whether at the [legal, political, organizational or technological levels](https://www.canada.ca/en/employment-social-development/corporate/reports/departmental-plan/2020-2021.html#h2.2-3.2), that impacts its ability to deliver on its mandate.
+Acquiring IT Solutions and Products can be accomplished using various methods and the acquisition is only a small part of the actual work required to be able to operate and maintain said technology.
+Adopting existing third-party solutions such as community-based open source software, or sometimes free software-as-a-service (SaaS), is an option often overlooked because of the lack of contractual support, warranties, indemnifications, etc.
+But those extra services are not part of the actual software, they are services that the government representatives have become used to bundle as part of the rights to use the software.
```
[the remark this answers is not in the document]
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:76]: Good catch, will adjust!

File: _pages/en/strategy-build-vs-buy.md [hunk 12]
```diff
@@ -67,48 +67,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
...
 
 This strategy complements existing IT initiatives (such as the [IITB Way Forward](http://dialogue/grp/IITB-DGIIT-Gov-New-Nouveau/Documents/Departmental%20IMIT%20Plan/19-20%20Plans/IITB%20Moving%20Forward%20v2.docx)) as well as supports them with new activities (see [Coherent set of actions](#coherent-set-of-actions)).
 
-This strategy's goal is to clarify the application of the GoC Mandatory Procedures for [Enterprise Architecture Assessments](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32602) and [on Application Programming Interfaces](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32604) when obtaining IT Solutions within ESDC IITB context, and provide a roadmap in getting to the adoption of this framework.
+This strategy's goal is to clarify the application of the GoC Mandatory Procedures for [Enterprise Architecture Assessments](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32602) and on [Application Programming Interfaces](https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=32604) when acquiring new or enhancing existing IT Solutions and IT products within ESDC IITB context, and provide actions items to enable the adoption of this guiding policy.
```
[the remark this answers is not in the document]
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:86]: Hmm, we changed it so that the purpose was less about putting a framework together (buy vs build) and more around action items, not a "plan with work breakdown structure and delivery timeframes"...

Might just remove the last part of the sentence then?

> [...],  and provide actions items to enable the adoption of this guiding policy.

File: _pages/en/strategy-build-vs-buy.md [hunk 13]
```diff
@@ -67,48 +67,60 @@ All ESDC personnel involved in the exploration of IT Solutions and in IT investm
...
 
 ## Guiding Policy
 
-The following policy reflects the decision adopted by the **CIO of ESDC** (approval by CIO not yet obtained) when assessing the most appropriate way of obtaining an IT solution, whether by adopting, buying or building it.
+The following policy reflects the decision adopted by the **CIO of ESDC** (approval by CIO not yet obtained) when assessing the most appropriate way of acquiring an IT solution, whether by adopting, buying or building it.
 Each policy statement is a declaration of that decision and has received the endorsement of its associated area of governance body (endorsements not yet obtained, see [Coherent set of actions](#coherent-set-of-actions)).
 
-This policy becomes active when an ESDC initiative requires that an IT Solution be obtained, whether delivered by ESDC or by a third-party provider.
+This policy becomes active when an ESDC initiative requires that an IT Solution be acquired, whether delivered by ESDC or by a third-party provider.
 Once active, all teams involved in the project, and the IT products involved in the IT Solution, must comply with this guiding policy.
```
[the remark this answers is not in the document]
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:96]: Actually, probably need to modify "project" with:

> This policy becomes active when an ESDC initiative requires that an IT Solution be acquired **or enhanced**, whether by ESDC or by a third-party provider.
> Once active, all teams involved in the **initiative**, and the IT products involved in the IT Solution, must comply with this guiding policy.

File: _pages/en/strategy-build-vs-buy.md [hunk 14]
```diff
@@ -132,31 +145,55 @@ Stakeholders are expected to still comply with existing policy instruments inclu
...
+
+### Specific to Application Programming Interfaces <!-- omit in toc -->
+
+- _Upcoming_
+
+### Specific to Acquisition of new or enhancements to IT Solutions and IT Products <!-- omit in toc -->
+
+#### Adopting <!-- omit in toc -->
+
+- Increased ratio of open source software solutions and products used in production compared to purchased or internally developed ones.
```
[the remark this answers is not in the document]
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:162]: That might be more accurate.

File: _pages/en/strategy-build-vs-buy.md (hunk 10 above)
Comment (gcharest) [reply _pages/en/strategy-build-vs-buy.md:435]: Yes, will update!

**Comment by gcharest** [comment]
Next iteration should increase connections to:
- Interoperability (increase opening of APIs as a mandatory set of requirements, no matter what tech is brought in or for any existing solutions to which changes will be introduced)
- Tech Debt (help decompose large monoliths into reusable structures - see Maryanne's blogs about how to do so)
- SDM (model to operate in a loosely coupled architecture), 
- Market Watch initiative (broadly looking at tech to match capabilities needs, taking ownership of tech choices vs choosing in pool of submitted tech)

Etc.

**Comment by gcharest** [comment]
Thanks a lot folks!
